### ISSUE TITLE
[1.12.2] Importing world save

### ISSUE THREAD

**Issue by JacobCube** [Cazsius/Spice-of-Life-Carrot-Edition#80]
I ran into a problem after importing my save to technic launcher.
https://paste.dimdev.org/apuvujozeh.mccrash

**Comment by juliand665** [comment]
Very weird. I think I've seen a similar crash before, but never went into it (#70). As such, let me restate what I said back then:

Thanks for the report! The log was a great start, but there's not much we can do without additional information. In order to fix a bug, the first step is always for us to reproduce it locally, and we can't do that based just on the information you've provided so far. Next steps:

- Does it happen for everyone on the server or just this one player?
- Does it still happen with just SoL: Carrot and AppleCore enabled? If not, which other mods need to be present for it to break?
(A good way to figure this out would be to remove chunks of mods until it stops breaking, at which point you've narrowed it down to that chunk having the incompatible mod. Rinse and repeat until you've singled out the cause.)

**Comment by srwaggon** [comment]
I am  now getting this crash.

Here's what I did to start receiving it:
I enabled 
```
    # Whether or not to reset the food list on death, effectively losing all bonus hearts.
    B:"Reset on Death"=false
```

to be true on both client side and serverside.
Then I restarted both.
I tried to connect to the server and received this stack trace

```
net.minecraft.util.ReportedException: Loading entity NBT
        at net.minecraft.entity.Entity.func_70020_e(Entity.java:1878) ~[vg.class:?]
        at net.minecraft.world.storage.SaveHandler.func_75752_b(SaveHandler.java:224) ~[bfb.class:?]
        at net.minecraft.server.management.PlayerList.func_72380_a(PlayerList.java:324) ~[pl.class:?]
        at net.minecraft.server.management.PlayerList.initializeConnectionToPlayer(PlayerList.java:116) ~[pl.class:?]
        at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher.completeServerSideConnection(NetworkDispatcher.java:255) ~[NetworkDispatcher.class:?]
        at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher.access$100(NetworkDispatcher.java:72) ~[NetworkDispatcher.class:?]
        at net.minecraftforge.fml.common.network.handshake.NetworkDispatcher$1.func_73660_a(NetworkDispatcher.java:205) ~[NetworkDispatcher$1.class:?]
        at net.minecraft.network.NetworkManager.func_74428_b(NetworkManager.java:285) ~[gw.class:?]
        at net.minecraft.network.NetworkSystem.func_151269_c(NetworkSystem.java:180) [oz.class:?]
        at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:790) [MinecraftServer.class:?]
        at net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:397) [nz.class:?]
        at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668) [MinecraftServer.class:?]
        at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526) [MinecraftServer.class:?]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_232]
Caused by: java.lang.NullPointerException
        at com.cazsius.solcarrot.tracking.ProgressInfo$ConfigInfo.isHearty(ProgressInfo.java:154) ~[ProgressInfo$ConfigInfo.class:1.12.2-1.8.3]
        at com.cazsius.solcarrot.tracking.ProgressInfo$ConfigInfo.shouldCount(ProgressInfo.java:150) ~[ProgressInfo$ConfigInfo.class:1.12.2-1.8.3]
        at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:174) ~[?:1.8.0_232]
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_232]
        at java.util.HashMap$KeySpliterator.forEachRemaining(HashMap.java:1556) ~[?:1.8.0_232]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_232]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_232]
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_232]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_232]
        at java.util.stream.LongPipeline.reduce(LongPipeline.java:461) ~[?:1.8.0_232]
        at java.util.stream.LongPipeline.sum(LongPipeline.java:419) ~[?:1.8.0_232]
        at java.util.stream.ReferencePipeline.count(ReferencePipeline.java:593) ~[?:1.8.0_232]
        at com.cazsius.solcarrot.tracking.ProgressInfo.<init>(ProgressInfo.java:34) ~[ProgressInfo.class:1.12.2-1.8.3]
        at com.cazsius.solcarrot.tracking.FoodList.updateProgressInfo(FoodList.java:117) ~[FoodList.class:1.12.2-1.8.3]
        at com.cazsius.solcarrot.tracking.FoodList.deserializeNBT(FoodList.java:69) ~[FoodList.class:1.12.2-1.8.3]
        at net.minecraftforge.common.capabilities.CapabilityDispatcher.deserializeNBT(CapabilityDispatcher.java:135) ~[CapabilityDispatcher.class:?]
        at net.minecraft.entity.Entity.func_70020_e(Entity.java:1852) ~[vg.class:?]
        ... 13 more
```

Then I tried a single player world and received the same crashlog as above.

I reset the property back to false on both sides, client and server.

I continue to receive the crash.

**Comment by srwaggon** [comment]
I can also confirm that I was able to resolve this issue by disabling the mod, relogging, then re-enabling the mod.

Unfortunately, my num-num progress had been lost but it seems like a small problem.

**Comment by juliand665** [comment]
Those logs/stack traces are nice, but I need to be able to reproduce the error locally, and I haven't been able to do so yet. Do you by chance have a world backup from before you fixed it, where it crashes upon loading?

**Comment by srwaggon** [comment]
@juliand665 Thanks for specifying what you need. I'm attaching a google drive link to a Twitch Client modded minecraft profile that you can import.

https://drive.google.com/file/d/1n8wmgVsnv6ud5NYrWe-nsV4N7WBeYHuj/view?usp=sharing

If you need any assistance with the Twitch Client profile import, I'm happily available, and can be reached via Discord. Just reach out to me and I'll find a way to give you a private link and we can setup a time.

When I log into this single player world, my client crashes with the above stacktrace. I have a minor concern that the crash is player dependent based on achievements. If you are able to load the world and play, then you aren't encountering the crash. In this event, I recommend following the steps in my previous comment to replicate the crash (flipping the setting `"Reset on Death"` to true after having gained hearts, then logging in again).

**Comment by juliand665** [comment]
Sorry for taking so long for this, but I finally got around to it! Thanks for the instance, @srwaggon; that worked great (and I found the easter egg I assume you put in :p). It turns out that for some reason you had wheat in your food list, and our code didn't expect an item in there to no longer be edible. Now when it encounters an inedible item it simply doesn't count it, but leaves it in in case it becomes edible again later. I'll be pushing out the update to fix this (1.8.4) very soon :)
